ecommerce/views.py

```diff
@@ -40,3 +39,114 @@ def update_profile(request):
         user_form = UserForm(instance=request.user)
         profile_form = ProfileForm(instance=request.user.profile)
     return render(request, 'ecommerce/profile.html', {'user_form': user_form,'profile_form': profile_form})
+
+@login_required
+@transaction.atomic
+def cart_add(request, pk):
+    if request.method == 'POST':
+        quantity = request.POST['quantity']
+    else:
+        quantity = 0
+
+    product_get = Product.objects.get(pk=pk)
+    if product_get:
+        product_stock = product_get.quantity
+    else:
+        product_stock = 0
+
+    in_cart = Cart.objects.filter(user=request.user, product__pk=pk)
+    if in_cart:
+        cart_quantity = in_cart.get(product__pk=pk).quantity
+    else:
+        cart_quantity = 0
+    
+    if int(quantity) > 0 :
+        if int(quantity) + cart_quantity > product_stock:
+            messages.add_message(request, messages.INFO, _("Out of stock limit"))
+        else:
+            product_temp = Cart.objects.filter(user=request.user, product__pk=pk)
+            if product_temp:
+                pd_temp = product_temp.get(product__pk=pk)
+                pd_temp.quantity = int(quantity) + cart_quantity
+                pd_temp.save()
+                messages.add_message(request, messages.INFO, _('You added a product to your cart'))
+            else:
+                product_new = Cart(user=request.user, product=Product.objects.get(pk=pk), quantity=quantity)
+                product_new.save()
+                messages.add_message(request, messages.INFO, _('You added a product to your cart'))
```

Comment:
lặp code message rồi e
```suggestion
            else:
                product_new = Cart(user=request.user, product=Product.objects.get(pk=pk), quantity=quantity)
                product_new.save()
            messages.add_message(request, messages.INFO, _('You added a product to your cart'))
```